### ISSUE TITLE
[libtorch]can not save a  vector<int> to AutogradContex->saved_data.

### ISSUE THREAD

**Issue by Mirocos** [pytorch/pytorch#71698]
### 🐛 Describe the bug

I want to save a `std::vector<int>` variable in cpp autograd forward.
So I write the code like this:
```c++
...forward(AutogradContext *ctx, ..., std::vector<int> diff_attrs_list){
    ctx->saved_data["diff_attrs_list] = diff_attrs_list;
}
```
but Exception throws at `jit_type.h` in `line 1592` `TORCH_CHECK.

So how to save a `vector<int>` variable in libtorch autograd?

### Versions

I download libtorch from this url:
https://download.pytorch.org/libtorch/nightly/cu115/libtorch-win-shared-with-deps-debug-latest.zip

**Comment by albanD** [comment]
Hi,

What is the error message?
Also does it work if you pass in a `std::vector<int64_t>` ?

**Comment by Mirocos** [comment]
Yes, when pass in a std::vector<int64_t> it works, but I thought it's better to provide a template entry
